cmd/adsysd/integration_tests/adsys_mount_test.go

```diff
@@ -0,0 +1,108 @@
+package adsys_test
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/adsys/internal/testutils"
+)
+
+func TestUserMountHandler(t *testing.T) {
+	fixtureDir := filepath.Join("testdata", t.Name())
+
+	env, target := setupBinaryForTests(t)
+
+	tests := map[string]struct {
+		mountsFile    string
+		sessionAnswer string
+		noKrbTicket   bool
+
+		wantStatus int
+	}{
+		// Single entries
+		"mount successfully nfs share": {mountsFile: "mounts_with_nfs_entry"},
+		"mount successfully smb share": {mountsFile: "mounts_with_smb_entry"},
+
+		// Anonymous entries
+		"mount successfully one anonymous nfs entry": {mountsFile: "mounts_with_anonymous_nfs_entry"},
+		"mount successfully one anonymous smb entry": {mountsFile: "mounts_with_anonymous_smb_entry"},
+
+		// Many entries
+		"mount successfully many entries with different protocols": {mountsFile: "mounts_with_many_entries"},
```

Comment:
I think the many entry with same protocols first is another use case so that we can see if the issue is due to many or just one protocol.